drivers/serial/uart_rtt.c

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2019 omSquare s.r.o.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/uart.h>
+#include <SEGGER_RTT.h>
+
+#if CONFIG_UART_RTT_DRIVER
+
+struct uart_rtt_config {
+	unsigned int channel;
+	const char *name;
+	void *up_buffer;
+	size_t up_size;
+	void *dw_buffer;
+	size_t dw_size;
+};
+
+static inline const struct uart_rtt_config *get_dev_config(struct device *dev)
+{
+	return dev->config->config_info;
+}
+
+static int uart_rtt_init(struct device *dev)
+{
+	/*
+	 * Channel 0 is initialized at compile-time, Kconfig ensures that
+	 * it is configured in correct, non-blocking mode. Other channels
+	 * need to be configured at run-time.
+	 */
+	if (get_dev_config(dev)) {
+		const struct uart_rtt_config *cfg = get_dev_config(dev);
+
+		SEGGER_RTT_ConfigUpBuffer(cfg->channel, cfg->name,
```

Comment:
Why not use `dev->config->name` instead of `cfg->name` (and remove this field from the `uart_rtt_config` structure)? Both these fields are pointing to the same string.

Reply:
Right, I overlooked it, thanks.